Add further explanation to external reading and writing to README

felippecaso opened the pull request:
Based on [this conversation](https://getdbt.slack.com/archives/C039D1J1LA2/p1677166441609539) in dbt Slack, add detailed explanation to `external_location` being able to be provided a explicit read function.

Resolves https://github.com/jwills/dbt-duckdb/issues/94 (hopefully)

jwills commented:
Hey @felippecaso, thanks so much for this! I thought this block should replace the one before it (that includes the simpler `read_parquet(...)` example), since it's substantially similar but more clearly demonstrates what you're allowed to do w/the `external_location` option on sources?

felippecaso commented:
> Hey @felippecaso, thanks so much for this! I thought this block should replace the one before it (that includes the simpler `read_parquet(...)` example), since it's substantially similar but more clearly demonstrates what you're allowed to do w/the `external_location` option on sources?

That makes sense! I've updated the text and added some other suggestions in text structure to address some of the points mentioned in https://github.com/jwills/dbt-duckdb/issues/94

felippecaso commented:
@jwills after reviewing the proposed changes in resolving conflicts, I feel the contributions are already covered by recent updates to `README`, so I'm closing this PR.

For the same reason, I also believe https://github.com/jwills/dbt-duckdb/issues/94 is already solved as well.
